src/GNDStk/Node/src/call-keywords.hpp

```diff
@@ -0,0 +1,159 @@
+
+// -----------------------------------------------------------------------------
+// Multi-query
+// -----------------------------------------------------------------------------
+
+// ------------------------
+// Private helpers
+// ------------------------
+
+private:
+
+// ()(tuple<...>)
+template<class... Ks>
+auto operator()(
+   const std::tuple<Ks...> &tup,
+   bool &found
+) GNDSTK_CONST {
+   log::debug(
+     "operator()(tuple<...>), tuple size == {}",
+      sizeof...(Ks)
+   );
+
+   bool head_found = false;
+   bool tail_found = false;
+
+   const auto ret = std::tuple_cat(
+      // Process tup's <0>
+      // We *don't* simply write:
+      //    std::make_tuple(operator()(std::get<0>(tup),head_found)),
+      // because doing so would decay references to non-references. That would
+      // mean, ultimately, that meta_t<void> and child_t<void> instances inside
+      // our keywords<...> objects, which are returned by our | (or) operators,
+      // would no longer produce [possibly const] *references* to std::string
+      // and Node, respectively, but would surreptitiously lose the references.
+      // We don't want to lose the references, though, because our multi-query
+      // system is far more awesome if we keep them.
```

Comment:
I agree, it's awesome ;-)

I can't wait to use it once the std::optional support is present as well.